packages/swapper/src/swappers/RelaySwapper/endpoints.ts

```diff
@@ -83,74 +63,50 @@ export const relayApi: SwapperApi = {
     from,
     stepIndex,
     tradeQuote,
-    supportsEIP1559,
-    assertGetEvmChainAdapter,
   }: GetUnsignedEvmTransactionArgs): Promise<EvmTransactionRequest> => {
     if (!isExecutableTradeQuote(tradeQuote)) throw Error('Unable to execute trade')
 
-    const relayQuote = tradeQuoteMetadata.get(tradeQuote.id)
-
-    if (!relayQuote) throw Error(`missing trade quote metadata for quoteId ${tradeQuote.id}`)
-
-    const swapSteps = relayQuote.steps.filter(step => step.id !== 'approve')
-
-    const { to, value, data, gas } = swapSteps[stepIndex].items?.[0].data ?? {}
-
-    const feeData = await evm.getFees({
-      adapter: assertGetEvmChainAdapter(chainId),
-      data,
-      to,
-      // This looks odd but we need this, else unchained estimate calls will fail with:
-      // "invalid decimal value (argument=\"value\", value=\"0x0\", code=INVALID_ARGUMENT, version=bignumber/5.7.0)"
-      value: bnOrZero(value.toString()).toString(),
-      from,
-      supportsEIP1559,
-    })
+    const { to, value, data, gas, maxFeePerGas, maxPriorityFeePerGas } =
+      tradeQuote.steps[stepIndex]?.relayTransactionMetadata ?? {}
+
+    if (
+      to === undefined ||
+      value === undefined ||
+      data === undefined ||
```

Comment:
q: is calldata guaranteed because of the ERC-20 associated send for Relay protocol fees?

Reply:
No

Calldata seems guaranteed because every EVM TXs are calling a smart contract (most of the time a relay receiver)

But as a security, we could kick the data safe guard as anyway if `to` and `value` is defined it should be enough to run the quote if either it has or hasn't data

Reply:
According to the docs, [sends exist](https://docs.relay.link/how-it-works/relay-protocol#direct-transfers).
Hence the question, ERC-20s will have a data field, but native tokens wouldn't.

Or to ask the other way around, are we doing sends?

Reply:
Closing as we don't do sends